storage/revision.go

```diff
@@ -32,7 +37,7 @@ func (a revision) GreaterThan(b revision) bool {
 }
 
 func newRevBytes() []byte {
-	return make([]byte, 8+1+8)
+	return make([]byte, revBytesLen)
```

Comment:
It's in performance path. I would allocate a byte for tombstone as well:
`make([]byte, revBytesLen, markedRevBytesLen)